lib/acts_as_taggable_on_mongoid/errors/duplicate_tag_error.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module ActsAsTaggableOnMongoid
+  module Errors
+    # Error raised if a Tag already exists but cannot be found appropriately when trying to create new tags.
+    class DuplicateTagError < StandardError
```

Comment:
Copied from here:
https://github.com/mbleigh/acts-as-taggable-on/blob/master/lib/acts-as-taggable-on.rb#L40

moved it into its own file.